### PR TITLE
Add Patient DSD Model In The Reports

### PR DESCRIPTION
**Pull request by solemabrothers**
The Appointment List needs more fields, i.e, 

- DSD Model

- Current Regimen

- Latest Vural Load Results and Date Taken

These will help us in the Community App  when serving clients.
The Card for this issue is here https://app.asana.com/0/1133167201254915/1188626194800633

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/ugandaemrreports/library/HIVPatientDataLibrary.java [hunk 1]
```diff
@@ -92,7 +89,7 @@ public PatientDataDefinition getCurrentViralLoad() {
     }
 
     public PatientDataDefinition getCurrentRegimen() {
-        return df.getObsByEndDate(hivMetadata.getCurrentRegimen(), null, TimeQualifier.LAST, df.getObsValueCodedConverter());
+        return df.getObsByEndDate(hivMetadata.getCurrentRegimen(), null, TimeQualifier.LAST, new ObsValueConverter());
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/library/HIVPatientDataLibrary.java:92]: @solemabrothers Why the change from df.getConverter to new Converter?

Comment (solemabrothers) [reply api/src/main/java/org/openmrs/module/ugandaemrreports/library/HIVPatientDataLibrary.java:92]: This is because the old converter would return Concept IDs instead of Values for Current Regimen in a CSV format. The new format works for both Excel and CSV. The same applies for dates

Comment (ssmusoke) [reply api/src/main/java/org/openmrs/module/ugandaemrreports/library/HIVPatientDataLibrary.java:92]: @solemabrothers so why not also remove the old convertor since its no longer used? So that the code is clean everywhere

File: api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpLostReport.java [hunk 2]
```diff
@@ -145,6 +145,9 @@ public ReportDefinition constructReportDefinition() {
         addColumn(dsd, "HIV Enrolled Date", hivPatientData.getEnrollmentDate());
         addColumn(dsd, "ART Start Date", hivPatientData.getArtStartDate());
         addColumn(dsd, "Current Regimen", hivPatientData.getCurrentRegimen());
+        addColumn(dsd, "VL Quantitative",  hivPatientData.getCurrentViralLoad());
```
Comment (metsprogram) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpLostReport.java:148]: @solemabrothers You need VL copies which is missing here

File: api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpLostReport.java [hunk 3]
```diff
@@ -145,6 +145,9 @@ public ReportDefinition constructReportDefinition() {
...
+        addColumn(dsd, "VL Date", hivPatientData.getViralLoadDate());
+        addColumn(dsd,"Model", hivPatientData.getDSDMModel());
```
Comment (metsprogram) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpLostReport.java:150]: @solemabrothers Call this "DSDM Model" instead of just model

File: api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpLostReport.java [hunk 4]
```diff
@@ -145,6 +145,9 @@ public ReportDefinition constructReportDefinition() {
         addColumn(dsd, "HIV Enrolled Date", hivPatientData.getEnrollmentDate());
         addColumn(dsd, "ART Start Date", hivPatientData.getArtStartDate());
         addColumn(dsd, "Current Regimen", hivPatientData.getCurrentRegimen());
+        addColumn(dsd, "VL Quantitative",  hivPatientData.getCurrentViralLoad());
+        addColumn(dsd, "VL Date", hivPatientData.getViralLoadDate());
```
Comment (metsprogram) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpLostReport.java:149]: @solemabrothers the DSDM Model enrollment date is not included

File: api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpLostReport.java (hunk 3 above)
Comment (solemabrothers) [reply api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpLostReport.java:150]: But @ssmusoke DSDM Model is repitition

**Comment by ssmusoke** [comment]
@solemabrothers The order of the columns for VL is important (especially for CSV), VL Date, VL Qualitative and VL Copies as well as DSDM Model and DSDM Model Enrollment Date

File: api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpLostReport.java (hunk 3 above)
Comment (ssmusoke) [reply api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpLostReport.java:150]: @solemabrothers it is repetition for clarity, as a person using this report for the first time is not sure what model you are referring to

**Comment by ssmusoke** [comment]
@solemabrothers

1. Change the title of the model column to "DSDM Model" not "DSD Model"
2. In Excel the DSD Model column has the word model not the actual DSDM model name 

CSV output is okay

**Comment by dbaluku** [comment]
@solemabrothers for your attention 
On Appointment Report
-DSD Model Enrollment Date dates are displayed as just numbers not dates 

On Missed Appointment Report

- Missing VL Qualitative column

- VL Quantitative displays dates 

- DSD Models not being displayed 

Lost Followup Report

- Current regimen is not being returned 

- VL Quantitative is returning a date 

- VL Qualitative column is missing

- DSD model is not being returned

- DSD Enrollment Date column is missing 

Lost Report

- VL Quantitative is returning a date 

- VL Qualitative column is missing

- DSD model is not being returned

- DSD Enrollment Date column is missing

**Comment by dbaluku** [comment]
@solemabrothers 

- Missed Appointment has two VL Quantitative column and one returns dates 

-Lost Clients and Lost to Followup Reports font is set to ten, please change it to 12 the standard to match other reports
